parser/src/parser/nodes.js

```diff
@@ -21,15 +21,13 @@ class Node extends Obj {
       closingTag: { start: false, end: false },
     };
     this.fields.forEach((field, i) => {
-      // The first two args are line/col numbers, so offset by 2
-      var val = arguments[i + 2];
```

Comment:
`arguments` is an object present in all functions that gives access to the args of the function ([MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments)).

The `arguments[i+2]` thing was causing the Node object to have incorrect values on the fields for some reason (it would set `this[field] = field` instead of `this[field] = val`). The `args[i]` thing I replaced it with should actually be equivalent, so I'm honestly not sure why it fixed it, but it did.

Reply:
Oh wow -- that's embarassing -- I actually did not know that. Thanks for the 101